webapp/src/components/surveyPost/surveyPost.tsx

```diff
@@ -0,0 +1,243 @@
+import TextQuestion from 'components/surveyPost/textQuestion/textQuestion';
+
+import './style.scss';
+
+import type {
+    CustomPostTypeComponentProps,
+    MattermostWindow,
+    SurveyResponse, UserSurvey,
+} from 'types/mattermost-webapp';
+
+function SurveyPost({post, isRHS}: CustomPostTypeComponentProps) {
+    const [survey, setSurvey] = useState<UserSurvey>();
+    const [disabled, setDisabled] = useState<boolean>(false);
+    const [expired, setExpired] = useState<boolean>(false);
+    const draftResponse = useRef<SurveyResponse>();
+    const [errorMessage, setErrorMessage] = useState<string>();
+
+    const linearScaleQuestionID = useRef<string>();
+
+    const fetchUserSurvey = async (surveyID: string) => {
+        // make actual API call to fetch survey here...
+        // Replace dummy data with data fetched from API call.
+        const survey: UserSurvey = {
+            surveyId: surveyID,
+            startDate: '01/04/2024',
+            endDate: '30/04/2024',
+            questions: [
+                {
+                    id: '49fc9a85-b4d8-424e-b13f-40344b168123',
+                    mandatory: true,
+                    system: true,
+                    text: 'How likely are you to recommend Mattermost?',
+                    type: 'linear_scale',
+                },
+                {
+                    id: 'aa026055-c97c-48d7-a025-c44590078963',
+                    mandatory: true,
+                    system: true,
+                    text: 'How can we make your experience better?',
+                    type: 'text',
+                },
+                {
+                    id: '0eee4429-5083-41ef-bda7-2ee9c5ece929',
+                    mandatory: false,
+                    system: false,
+                    text: 'option question text',
+                    type: 'text',
+                },
+            ],
+            status: 'in_progress',
+
+            // response: {
+            //     responses: {
+            //         '49fc9a85-b4d8-424e-b13f-40344b168123': '8',
+            //         'aa026055-c97c-48d7-a025-c44590078963': 'Response 1',
+            //         '0eee4429-5083-41ef-bda7-2ee9c5ece929': 'Response 2',
+            //     },
+            //     dateCreated: format(new Date(), 'dd/MM/yyyy'),
+            // },
+        };
+
+        setSurvey(survey);
+
+        // the first linear scale question is the system default rating question
+        const linearScaleQuestion = survey.questions.find((question) => question.type === 'linear_scale');
+        linearScaleQuestionID.current = linearScaleQuestion?.id;
+
+        const responsesExist = survey.response !== undefined;
+        const surveyExpired = survey.status === 'ended';
+
+        setDisabled(responsesExist || surveyExpired);
+        setExpired(surveyExpired);
+    };
+
+    // fetch data on initial mount
+    useEffect(() => {
+        fetchUserSurvey(post.props.surveyID);
```

Comment:
I thought about this but realistically there shouldn't be a large number of survey posts. I'm thinking about storing the response in the post props itself on submission so it doesn't need to make those API calls, but yeah, something that we could do. But I don't think its a concern.

Reply:
The part I'm a bit worried about is that the API calls are made every time you switch to the DM.

Reply:
That's unfortunately something that we don't have a general solution for yet. For stuff in Redux, we assume that objects are always kept in sync via the websocket or we assume that they're always stale and need to be refetched like this.

I agree that it's probably fine in this case though since it only affects that one DM channel